src/script/cryptography/CryptographyMapper.js

```diff
@@ -234,23 +233,21 @@ export class CryptographyMapper {
   }
 
   _mapAvailability(availability) {
+    const knownAvailabilityTypes = [
+      Availability.Type.NONE,
+      Availability.Type.AVAILABLE,
+      Availability.Type.AWAY,
+      Availability.Type.BUSY,
+    ];
+
+    if (!knownAvailabilityTypes.includes(availability.type)) {
+      const message = 'Unhandled availability type';
+      throw new z.error.CryptographyError(z.error.CryptographyError.TYPE.UNHANDLED_TYPE, message);
+    }
+
     return {
       data: {
-        availability: (() => {
-          switch (availability.type) {
-            case Availability.Type.NONE:
-              return AvailabilityType.NONE;
-            case Availability.Type.AVAILABLE:
-              return AvailabilityType.AVAILABLE;
-            case Availability.Type.AWAY:
-              return AvailabilityType.AWAY;
-            case Availability.Type.BUSY:
-              return AvailabilityType.BUSY;
-            default:
-              const message = 'Unhandled availability type';
-              throw new z.error.CryptographyError(z.error.CryptographyError.TYPE.UNHANDLED_TYPE, message);
-          }
-        })(),
+        availability: availability.type,
```

Comment:
I believe the types are different here.

`Availability.Type` vs `AvailabilityType`.

Do they resolve to the same values?

Reply:
Yes, they are the same:

https://github.com/wireapp/wire-webapp/blob/c13f05233c2a4e0b1a1d4dfc2ced055ad3725a0b/src/script/user/AvailabilityType.ts#L20-L24

[`Protobuf.d.ts`](https://unpkg.com/@wireapp/protocol-messaging@1.24.1/web/Protobuf.d.ts):
```ts
enum Type {
    NONE = 0,
    AVAILABLE = 1,
    AWAY = 2,
    BUSY = 3
}
```